src/ImageSharp/Formats/Jpeg/GolangPort/Components/Decoder/InputProcessor.cs

```diff
@@ -107,6 +107,13 @@ public byte ReadByte()
             return this.Bytes.ReadByte(this.InputStream);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public OrigDecoderErrorCode ReadByteUnsafe(out byte result)
```

Comment:
Ah that's better! 👍

Reply:
Actually defaulting to this logic would be the best, but I felt it's more safe to continuously refactor stuff towards it, also keeping the exception throwing where we haven't yet implemented the "smart" error management.